packages/features/auth/lib/scopeValidator.ts

```diff
@@ -0,0 +1,55 @@
+import { AccessScope } from "@calcom/prisma/enums";
+
+export type ScopeValidationResult =
+  | { success: true; scopes: AccessScope[] }
+  | { success: false; error: string };
+
+/**
+ * Parses and validates OAuth scopes against the AccessScope enum.
+ * Supports both space-delimited (OAuth spec) and comma-delimited formats.
+ * 
+ * @param input - Scope string or array to validate
+ * @param options - Validation options
+ * @returns Validation result with sanitized scopes or error
+ */
+export function parseAndValidateScopes(
+  input: string | string[],
+  options?: { allowEmpty?: boolean }
+): ScopeValidationResult {
+  let scopeArray: string[];
+  if (typeof input === "string") {
+    scopeArray = input.split(/[\s,]+/).filter(Boolean);
+  } else {
+    scopeArray = input;
+  }
+
+  const normalized = Array.from(new Set(scopeArray.map((s) => s.trim().toUpperCase()))).sort();
+
+  if (normalized.length === 0) {
+    if (options?.allowEmpty === false) {
+      return { success: false, error: "At least one scope is required" };
+    }
+    return { success: true, scopes: [AccessScope.READ_PROFILE] };
+  }
```

Comment:
This part raises some flags for me. In the places this is being called we're not passing `allowEmpty` and should we really be returning `success` and a scope if no scopes were provided?